lib/loader/loader.js

```diff
@@ -217,6 +217,11 @@ var loader, define, requireModule, require, requirejs;
   };
 
   define = function(name, deps, callback) {
+    var module = registry[name];
+    if (module && module.state !== 'new' && module.state !== 'errored') {
```

Comment:
Agreed, the previous module state is `reifying` but my point is you are most likely going to change the module behavior between two requires if you allow redefine.

We should consider the behavior of the module as `locked` once it enters `pending` state.